Traceback (most recent call last):
  File "/opt/anaconda3/envs/jupyter-book-py311/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/anaconda3/envs/jupyter-book-py311/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/jupyter-book-py311/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 173, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/jupyter-book-py311/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/jupyter-book-py311/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/anaconda3/envs/jupyter-book-py311/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/anaconda3/envs/jupyter-book-py311/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# folium„Éû„ÉÉ„Éó„Å´GEE„ÇíË°®Á§∫„Åï„Åõ„ÇãÈñ¢Êï∞
def add_ee_layer(self, ee_image_object, vis_params, name):
    map_id_dict = ee.Image(ee_image_object).getMapId(vis_params)
    folium.raster_layers.TileLayer(
        tiles=map_id_dict['tile_fetcher'].url_format,
        attr='<a href="https://earthengine.google.com/">Google Earth Engine</a>',
        name=name,
        overlay=True,
        control=True
    ).add_to(self)

# Âú∞Âõ≥„Çí‰ΩúÊàê
fmap = folium.Map(location=center, zoom_start=10)

basemaps['Google Maps'].add_to(fmap)
basemaps['Google Satellite Hybrid'].add_to(fmap)

# Sentinel-1
sentinel1 = ee.ImageCollection('COPERNICUS/S1_GRD') \
    .filterDate('2020-01-01', '2023-12-31') \
    .filter(ee.Filter.listContains('transmitterReceiverPolarisation', 'VV')) \
    .filter(ee.Filter.listContains('transmitterReceiverPolarisation', 'VH')) \
    .filter(ee.Filter.eq('instrumentMode', 'IW'))
# ImageCollection„Åã„Çâ„É¢„Ç∂„Ç§„ÇØ„Çí‰ΩúÊàê
mosaic = sentinel1.mosaic()

# Âú∞Âõ≥„Å´„É¨„Ç§„É§„Éº„ÇíËøΩÂä†
folium.Map.add_ee_layer = add_ee_layer
fmap.add_ee_layer(mosaic.select(['VH']), {'min': -25,'max': 5}, 'Sentinel-1 VH')
fmap.add_ee_layer(mosaic.select(['VV']), {'min': -25,'max': 5}, 'Sentinel-1 VV')
# „Ç≥„É≥„Éà„É≠„Éº„É´ÈÉ®ÂìÅ„ÇíËøΩÂä†
fmap.add_child(folium.LayerControl(collapsed = False).add_to(map))
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
Cell [0;32mIn[3], line 32[0m
[1;32m     30[0m fmap[38;5;241m.[39madd_ee_layer(mosaic[38;5;241m.[39mselect([[38;5;124m'[39m[38;5;124mVV[39m[38;5;124m'[39m]), {[38;5;124m'[39m[38;5;124mmin[39m[38;5;124m'[39m: [38;5;241m-[39m[38;5;241m25[39m,[38;5;124m'[39m[38;5;124mmax[39m[38;5;124m'[39m: [38;5;241m5[39m}, [38;5;124m'[39m[38;5;124mSentinel-1 VV[39m[38;5;124m'[39m)
[1;32m     31[0m [38;5;66;03m# „Ç≥„É≥„Éà„É≠„Éº„É´ÈÉ®ÂìÅ„ÇíËøΩÂä†[39;00m
[0;32m---> 32[0m fmap[38;5;241m.[39madd_child(folium[38;5;241m.[39mLayerControl(collapsed [38;5;241m=[39m [38;5;28;01mFalse[39;00m)[38;5;241m.[39madd_to([38;5;28mmap[39m))

File [0;32m/opt/anaconda3/envs/jupyter-book-py311/lib/python3.11/site-packages/branca/element.py:136[0m, in [0;36mElement.add_to[0;34m(self, parent, name, index)[0m
[1;32m    134[0m [38;5;28;01mdef[39;00m [38;5;21madd_to[39m([38;5;28mself[39m, parent, name[38;5;241m=[39m[38;5;28;01mNone[39;00m, index[38;5;241m=[39m[38;5;28;01mNone[39;00m):
[1;32m    135[0m [38;5;250m    [39m[38;5;124;03m"""Add element to a parent."""[39;00m
[0;32m--> 136[0m     parent[38;5;241m.[39madd_child([38;5;28mself[39m, name[38;5;241m=[39mname, index[38;5;241m=[39mindex)
[1;32m    137[0m     [38;5;28;01mreturn[39;00m [38;5;28mself[39m

[0;31mAttributeError[0m: type object 'map' has no attribute 'add_child'

